src/main.ts

```diff
@@ -30,18 +30,36 @@ async function main() {
         let servicePrincipalKey = secrets.getSecret("$.clientSecret", true);
         let tenantId = secrets.getSecret("$.tenantId", false);
         let subscriptionId = secrets.getSecret("$.subscriptionId", false);
+        let resourceManagerEndpointUrl = secrets.getSecret("$.resourceManagerEndpointUrl", false);
+        let profileVersion = secrets.getSecret("$.profileVersion", false);
+        let customEnvironmentName = secrets.getSecret("$.customEnvironmentName", false);
         const enableAzPSSession = core.getInput('enable-AzPSSession').toLowerCase() === "true";
         if (!servicePrincipalId || !servicePrincipalKey || !tenantId || !subscriptionId) {
             throw new Error("Not all values are present in the creds object. Ensure clientId, clientSecret, tenantId and subscriptionId are supplied.");
         }
         // Attempting Az cli login
+        if (customEnvironmentName != "") {
+            console.log(`Registering custom cloud: "${customEnvironmentName}" with ARM endpoint: "${resourceManagerEndpointUrl}"`);
+            try {
+                await executeAzCliCommand(`cloud register -n "${customEnvironmentName}" --endpoint-resource-manager "${resourceManagerEndpointUrl}" `, false);

```

Comment:
Kevault endpoint suffix and storage endpoint suffix are needed to be passed in, can we make them as parameters as well?

Reply:
or KV and storageendpoints are not needed for the operations being performed here?

Reply:
added keyvault and storage endpoints suffixes during cloud registration